test/plugin/base_test.rb

```diff
@@ -1608,8 +1660,16 @@ def verify_json_log_entries(n, params, payload_type = 'textPayload')
         labels ||= request['labels']
         labels.merge!(entry['labels'] || {})
 
-        assert_equal \
-          "projects/#{params[:project_id]}/logs/#{params[:log_name]}", log_name
+        # For multi tag requests, each log name should match the index.
+        if params[:log_name] == 'multi'
+          assert_equal \
+            "projects/#{params[:project_id]}/logs/tag#{i}",
```

Comment:
Hmm, why can't you check this in a separate loop for requests instead of inventing a special log name? You already check the entry-level `log_name` in the block.

Reply:
In fact, the complexity comes when I was trying to check both log names at the request level and the entry level. On a second thought, since entry level `log_name` always take precedence of the request level one, maybe it's better to check the `combined` log name instead (That is how `def verify_log_entries` has been doing it).

I just refactored the tests to simplify log name checks. I put it in a separate commit just to emphasize.